src/Symfony/Component/DependencyInjection/Compiler/ValidateEnvPlaceholdersInExtensionConfigPass.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Definition\BaseNode;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\EnvPlaceholderParameterBag;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+
+/**
+ * Validates environment variable placeholders used in extension configuration with dummy values.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class ValidateEnvPlaceholdersInExtensionConfigPass implements CompilerPassInterface
+{
+    private static $typeFixtures = array('array' => array(), 'bool' => false, 'float' => 0.0, 'int' => 0, 'string' => '');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!class_exists(BaseNode::class) || !($extensions = $container->getExtensions())) {
+            return;
+        }
+
+        $resolvingBag = $container->getParameterBag();
+        if (!$resolvingBag instanceof EnvPlaceholderParameterBag) {
+            return;
+        }
+
+        $defaultBag = new ParameterBag($container->getParameterBag()->all());
+        $envTypes = $resolvingBag->getProvidedTypes();
+        foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+            $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+            if (!($types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string'))) {
+                continue;
+            }
+            $default = ($hasEnv = (false === $i && $defaultBag->has("env($env)"))) ? $defaultBag->get("env($env)") : null;
+//            if (null !== $default && !in_array($type = gettype($default), $types, true)) {
+//                throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
```

Comment:
ok. needs to happen in fact; looking at the test i comment out :) but we can move the check to the bag itself no?

Reply:
you mean that we should enforce that types returned by `getProvidedTypes` must be something that `gettype()` could return? if yes, then the check should be added on 3.4, so that nobody relies on it

Reply:
i just think `ValidateEnvPlaceholdersInExtensionConfigPass` should not need to validate the parameter bag itself :thinking: